### PR TITLE
Improve documentation of packaging build targets

### PR DESCRIPTION
**Pull request by shonfeder**
@soareschen hit some snags when writing the nix derivation for Aapalche for `cosmos.nix`. It seemed the root cause was unclear documentation about the intended build target for packaging enduser releases.

These documentation tweaks are meant to improve that situation.

Tagging @soareschen in for review: please let me know if you think this will help people who might end up in that situation in the future!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# a good old Makefile for the end users, so they don't need to learn SBT commands
+# A good old Makefile for the end users, so they don't need to learn SBT commands
```
Comment (shonfeder) [inline Makefile:1]: unrelated cleanup of capitalization

File: Makefile [hunk 2]
```diff
@@ -48,8 +50,6 @@ test/tla/%.md: target/test/tla/%.md.corrected
 	cp -f $< $@
 
 fmt-check:
-  # TODO: rm if we decide to keep running on all source files
-	# git fetch origin
```
Comment (shonfeder) [inline Makefile:52]: unrelated removal of dead code and outdated todo

**Review by soareschen — APPROVED** [review]

**Review by thpani — APPROVED** [review]
👍
